### DISCUSSION TITLE
How to multiply spectrum with an "emissive light spectrum"?

### DISCUSSION THREAD

**Discussion by ademmler** [colour-science/colour#899] [Q&A] [answered]
Hi,

this are my first steps and tries in Colour - forgive me if I am doing fully wrong ;-)

How can I multiply a given spectrum of a color with a "spectrum of a light source"?

`...
sd_blue = colour.SpectralDistribution(data_blue, name='Blue')
sd_light = colour.SDS_ILLUMINANTS['B'] / 100
sd = sd_blue * sd_light
print(repr(sd))
`
I get this error:
RuntimeWarning: invalid value encountered in less_equal
  V = np.where(L <= 0.0031308, L * 12.92, 1.055 * spow(L, 1 / 2.4) - 0.055)

As far as I see the spectral data does not match. 
Do I have to use shape and if yes how to get them ready for multiplication?

**Comment by thomasmansencal** [comment, marked as answer]
Hi @ademmler,

The warning you get is probably related to a different section of the code, it is coming from the [colour.models.eotf_inverse_sRGB](https://github.com/colour-science/colour/blob/3cd6ab8d4c3483bcdeb2d7ef33967160808c0bb2/colour/models/rgb/transfer_functions/srgb.py#L86) definition.

Multiplying two spectral distributions should be straightforward: You simply multiply one with the other! There is one important thing to consider though: Whether their spectral shapes are the same.

- If the spectral shapes are differents, you will get `NaNs` where the bins do not match.
- In that case, you can leverage the fact that the `colour.SpectralDistribution` is built on top of a continuous signal representation and thus evaluate that function on the wavelengths of interest.
- Alternatively, you can align the spectral distributions together before the operation:

```python
In [1]: import colour

In [2]: a = colour.sd_ones(colour.SpectralShape(400, 700, 10))

In [3]: b = colour.sd_ones(colour.SpectralShape(400, 700, 20)) * 2

In [4]: (a * b).values
Out[4]:
array([  2.,  nan,   2.,  nan,   2.,  nan,   2.,  nan,   2.,  nan,   2.,
        nan,   2.,  nan,   2.,  nan,   2.,  nan,   2.,  nan,   2.,  nan,
         2.,  nan,   2.,  nan,   2.,  nan,   2.,  nan,   2.])

In [5]: (a * b[a.wavelengths]).values
Out[5]:
array([ 2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,
        2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,
        2.,  2.,  2.,  2.,  2.])

In [6]: (a * b.copy().align(a.shape)).values
Out[6]:
array([ 2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,
        2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,
        2.,  2.,  2.,  2.,  2.])
```

**Comment by ademmler** [comment]
Thx for your help!
